src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -407,12 +422,21 @@ private void GenerateMetadataNodes()
                         // we just pull in methods that have attributes on them.
                         if ((method.Attributes & MethodAttributes.MemberAccessMask) == MethodAttributes.Public &&
                             (method.Attributes & MethodAttributes.Static) != 0 &&
+                            method.GetParameters().Count > 0 &&
                             method.GetCustomAttributes().Count > 0)
                         {
-                            var definition = new MetadataDefinition(
-                                MetadataDefinitionKind.Member, _metadataReader.GetString(method.Name));
-
-                            definitionMap.Add(definition.Name, definition);
+                            // Decode method signature to get the target type name (i.e. type name for the first parameter)
+                            var blob = _metadataReader.GetBlobReader(method.Signature);
+                            var decoder = new SignatureDecoder<ParameterTypeInfo, object>(ParameterTypeInfoProvider.Instance, _metadataReader, genericContext: null);
+                            var signature = decoder.DecodeMethodSignature(ref blob);
+
+                            if (signature.ParameterTypes.Length > 0)
```

Comment:
:(   can you add a comment here that says it would be good to not have to allocate all of these.  the point of using the direct metadata reader code here was to try to avoid allocations as much as possible.

Reply:
Added. By the way I thought about copying code from metadata reader as Tomas mentioned, but eventually give up since I think it's a bad idea. We need a proper API to do this.

Reply:
works for me.  #resolved